### PR TITLE
Add an option to pass in promise module from the Monocle constructor

### PR DESCRIPTION
**Pull request by marianeaj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,3 +64,28 @@ app.config(function(monocleProvider) {
...
+
+var monocleProvider = new monocle(nodeAdapter);
+
+// Set the host path for API calls.
+// All API calls will automatically be mounted onto this host path.
+monocleProvider.setHost('http://www.yourhost.com');
+
+// Set the base path for API calls.
+// All API calls will automatically be mounted onto this base path.
+monocleProvider.setBase('/api/v2');
```
Comment (djvirgen) [inline README.md]: Can you throw in an example `get` or `post` request?

File: lib/monocle.js [hunk 2]
```diff
@@ -5,20 +5,27 @@ var ResourceCache = require('./cache/resource');
...
 
-function Monocle(http) {
+function Monocle(http, promise) {
     this._http = http;
     this._base = '/';
     this._cache = new Store(new ResourceCache('monocle', { capacity: 100 }));
     this._batched = [];
     this._batchTimeout = null;
     this._queuedGets = {};
+    this._promise = Promise || promise;
```
Comment (djvirgen) [inline lib/monocle.js]: I think this should be the other way around. If you inject `promise`, it should replace the default promise implementation.

File: test/unit/lib/monocle_test.js [hunk 3]
```diff
@@ -42,6 +42,14 @@ describe('Monocle API Client', function() {
                     this.http.request.calledWith(method.toUpperCase(), '/my-base/foo').should.be.true;
                 });
 
+                it('prepends host if provided', function() {
```
Comment (djvirgen) [inline test/unit/lib/monocle_test.js]: For the node adapter, should host be required?

Comment (marianeaj) [inline test/unit/lib/monocle_test.js]: it is required ... right now if you don't do setHost the API will return an error

Comment (djvirgen) [inline test/unit/lib/monocle_test.js]: Cool, is there a test for this?

Comment (marianeaj) [inline test/unit/lib/monocle_test.js]: in node_tests there is a test that checks whether errors are rejected properly.... if we don't set a host or set a non existing host for example this request will return an error and we are stubbing this with this test

**Comment by djvirgen** [comment]
🍒

**Pull request merged**